.github/workflows/pr.yml

```diff
@@ -131,3 +128,6 @@ jobs:
           go-version: ${{ env.GO_VERSION }}
       - name: Check generated files are up to date
         run: make fast_check_generated
+        env:
+          # needed for github-action-config.json generation
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
this decreases the readability: the token's scope is for the job.

Reply:
Maybe. On the other hand, the token is a security sensitive thing, and it seems prudent to make it available to as few steps as possible. In a sense, the scope of the token can be said to be the step(s) of a job where it's needed, only one here that I can tell. But sure, can revert this one if you want.

What about the corresponding change in tag.yml, it contained the same env/secret setting both on the job and the step level where it's needed? I removed the apparently redundant job level one.